### PR TITLE
Feat: Radio button

### PR DESCRIPTION
**Pull request by andoulla**
Added a `RadioButton` and a `RadioButtonGroup` that implements the structure of Radix Radio component.

image

### CODE REVIEW

File: sandbox/index.tsx [hunk 1]
```diff
@@ -3,11 +3,32 @@ import * as ReactDOM from 'react-dom'
...
 import { Box, globalCss } from '../dist'
+import { RadioButton } from '../src/components/radio/RadioButton';
+import { RadioButtonGroup } from '../src/components/radio/RadioButtonGroup';
 
 globalCss(reset)()
 
 const App = () => {
-  return <Box />
+  return (
+    <Box
```
Comment (andoulla) [inline sandbox/index.tsx:13]: I will revert these changes it was by mistake

File: src/components/radio/RadioButton.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  },
+  ':checked + &': { backgroundColor: '$secondary300' },
+  ':disabled + &': {
+    backgroundColor: '$tonal600',
+    border: '2px solid $tonal600'
+  }
+})
+
+const StyledIndicator = styled(RadioGroup.Indicator, {
+  width: 6,
```
Comment (andoulla) [inline src/components/radio/RadioButton.tsx:33]: we don't have a value of six in the theme, so not sure if we should add it

Comment (thomasdigby) [reply src/components/radio/RadioButton.tsx:33]: Could we use 8? We can tweak the theme to add an extra value to the sizing scale

Comment (andoulla) [reply src/components/radio/RadioButton.tsx:33]: Btw this is how it would look if using 8 
image

Comment (thomasdigby) [reply src/components/radio/RadioButton.tsx:33]: Yeah we haven't used a sizing scale before so the current values are temporary. I think 8/16/24/32/48/64 could work, let's just hard code 8 for now

File: src/components/radio/RadioButtonGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import * as RadioGroup from '@radix-ui/react-radio-group'
+import * as React from 'react'
+
+export const RadioButtonGroup: React.FC = ({ children }) => (
```
Comment (hollg) [inline src/components/radio/RadioButtonGroup.tsx:4]: Is there any reason not to simply export `RadioGroup.Root`, allowing our users to pass whatever props they want to that component itself? This current version doesn't allow for styling of the group etc

File: src/components/radio/RadioButtonGroup.test.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { render, screen } from '@testing-library/react'
+import React from 'react'
+
+import { RadioButtonGroup } from '.'
+
+describe(`RadioButton Groupcomponent`, () => {
```
Comment (hollg) [inline src/components/radio/RadioButtonGroup.test.tsx:6]: Absolutely tiny point, but we're missing a space between `Group` and `component` in the test name

File: src/components/radio/RadioButton.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  height: '$0',
+  display: 'inline-flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  verticalAlign: 'middle',
+  '&:focus': {
+    outline: 'none'
+  },
+  ':checked + &': { backgroundColor: '$secondary300' },
+  ':disabled + &': {
```
Comment (thomasdigby) [inline src/components/radio/RadioButton.tsx:26]: This should be `[disabled]` and our nested selectors require `&` prefix now, so: `&[disabled] + &`

Worth applying the `&` to any other selectors that need it

File: src/components/radio/RadioButtonGroup.test.tsx (hunk 4 above)
Comment (andoulla) [reply src/components/radio/RadioButtonGroup.test.tsx:6]: Thanks

File: src/components/radio/RadioButton.tsx (hunk 5 above)
Comment (andoulla) [reply src/components/radio/RadioButton.tsx:26]: I switched if for the disabled selector, however, the others don't seem to work with the attribute selector so I will leave them.

File: src/components/radio/RadioButtonGroup.tsx (hunk 3 above)
Comment (andoulla) [reply src/components/radio/RadioButtonGroup.tsx:4]: Now that I think about it again, not there isn't. I'm going to wrap this in styled so we get the styling options and simplify the component as you suggested

**Review by thomasdigby — APPROVED** [review]

**Pull request merged**